### PR TITLE
Added custom properties

### LINKED ISSUE #293: Adding customProperties to choices through markup
Is it possible to add customProperties to choices for select-multiple through markup?
Like through data properties or something?

### PR DESCRIPTION
**Pull request by julian-kuroiwa**
## Description
Added custom properties to the Choices options
This Close #293.

## Motivation and Context
It was required to solve a problem that i was having search. Now when i search for words that i have on custom properties it shows as an option.

## How Has This Been Tested?
Tested on local environment.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by julian-kuroiwa** [comment]
@jshjohnson can you take a look, pls?

File: src/scripts/lib/utils.js [hunk 1]
```diff
@@ -229,6 +229,8 @@ export const dispatchEvent = (element, type, customArgs = null) => {
   return element.dispatchEvent(event);
 };
 
+export const parseJSON = (s) => (!s || typeof JSON.parse !== 'function') ? {} : JSON.parse(s.replace(/'/g, '"'));
```
Comment (jshjohnson) [inline src/scripts/lib/utils.js]: Can the argument be more descriptive please :) also the JSON.parse shouldn't be necessary?

Comment (julian-kuroiwa) [reply src/scripts/lib/utils.js]: PR updated, maybe now its fine, simple and it does not requires parseJSON function :)

**Comment by julian-kuroiwa** [comment]
@jshjohnson can you see if its ok?

File: src/scripts/choices.js [hunk 2]
```diff
@@ -1907,6 +1907,9 @@ class Choices {
           selected: o.selected,
           disabled: o.disabled || o.parentNode.disabled,
           placeholder: o.hasAttribute('placeholder'),
+          customProperties: {
+            description: o.getAttribute('data-custom-properties'),
```
Comment (jshjohnson) [inline src/scripts/choices.js]: What's the reason for the `description` key?

File: src/scripts/lib/utils.js (hunk 1 above)
Comment (jshjohnson) [inline src/scripts/lib/utils.js]: This is no longer being used

**Comment by jshjohnson** [comment]
Hey @julian-kuroiwa - I think you applied your changes against the develop branch, not the master branch. The develop branch is stale - might be worth rebasing from master.

**Pull request closed without merging**